src/leer.java

```diff
@@ -8,264 +8,210 @@
-		           U=calculate.haversine(lat, lng, lat1, lng1, id_ride, timestamp, timestamp1,i);
-				   //calling filter
-		           campos=filter(campos,U);	   
-				   //System.out.println(Arrays.toString(campos));
-				   //System.out.println(U);
-				   
+		           String[] fields_line2 = line.split(",");
+		           lat_line2 = fields_line2[1];
+		           lng_line2 = fields_line2[2];
+		           timestamp_line2 = fields_line2[3];
+		           U=calculate.calculate_speed(lat, lng, lat_line2, lng_line2, id_ride, timestamp, timestamp_line2,i);
+				   filter(fields_line1,U);
 				   i++;
-	   }   
-	 }
-	 catch (IOException e) {
-	  e.printStackTrace();
-	 }
-	 finally {
-		 //calling valor to print answer
-	  result= calculate.valor(i,id_ridef,fare);    
-		 
-	  if (bufferLectura != null) {
-	   try {
-	    bufferLectura.close();
-	   } 
-	   
-	   catch (IOException e) {
-	    e.printStackTrace();
-	   }
-	  }
-	 
-	  try {
-          String ruta = "/Users/a.manning/Downloads/result.txt";
-          String contenido = result;
-            
-          File file = new File(ruta);
-          // if the file doesn't exists is created
-          if (!file.exists()) {
-              file.createNewFile();
-          }
-          FileWriter fw = new FileWriter(file);
-          BufferedWriter bw = new BufferedWriter(fw);
-          
-		bw.write(contenido);
-          bw.close();
-      } catch (Exception e) {
-          e.printStackTrace();
-      }
-  }
-	 } 
-     	 	
-
-
-
-
-public  double haversine (String lat,String lng, String lat1,String lng1, String id_ride, String timestamp, String timestamp1,int i){
- 
-	double a,c,distance,U,deltlat,deltlng,delttime;
+	            }
+	    }
+	    catch (IOException e) {
+	          e.printStackTrace();
+	    }
+	    finally {
+	        result= calculate.prepare_result(i,id_ride_final,fare);
+		    if (buffer_read != null) {
+	           try {
+	               buffer_read.close();
+	           }
+	           catch (IOException e) {
+	                e.printStackTrace();
+			   }
+	        }
+		    try {
+                String route = "/Users/a.manning/Downloads/result.txt";
+                String content = result;
+			    File file = new File(route);
+                // if the file doesn't exist is created
+                if (!file.exists()) {
+                   file.createNewFile();
+                }
+			    FileWriter fw = new FileWriter(file);
+                BufferedWriter bw = new BufferedWriter(fw);
+		        bw.write(content);
+                bw.close();
+            }
+		    catch (Exception e) {
+                  e.printStackTrace();
+			}
+        }
+	}
+public  double calculate_speed (String lat,
+						  String lng,
+						  String lat_line2,
+						  String lng_line2,
+						  String id_ride,
+						  String timestamp,
+						  String timestamp_line2,
+						  int i)
+{
+	double a,c,distance,U,delta_latitude,delta_longitude,delta_time;
 	int R =6371;
```

Comment:
Another magic number, what is 6371?

Reply:
This number is the earth radius, would it be better if i add a comment about this number? :)